client/packages/inventory/src/Stocktake/api/hooks/document/useInsertStocktake.ts

```diff
@@ -12,11 +13,22 @@ export const useInsertStocktake = () => {
   return useMutation<
```

Comment:
does this work ? 
```
export const useInsertStocktake = () => {
  const queryClient = useQueryClient();
  const navigate = useNavigate();
  const api = useStocktakeApi();
  return useMutation(
    api.insertStocktake,
    {
      onSuccess: ({ stocktakeNumber }) => {
        navigate(String(stocktakeNumber));
        return queryClient.invalidateQueries(api.keys.base());
      },
    }
  );
};

```

Reply:
> does this work ?

yep, thanks! there were issues previously with ts annotations being required, though with some of the later updates, the inferring is working better.

Reply:
though it requires adding the object and deref in the api.